### DISCUSSION TITLE
3D Near-to-Far-Field-Transformation with periodic boundary conditions?

### DISCUSSION THREAD

**Discussion by Snens98** [NanoComp/meep#2946] [Q&A] [answered]
Dear all,

I am currently working on implementing a periodic simulation using MEEP and would appreciate your insights on a specific issue I’m encountering. The model I am working with consists of a cylindrical meta-lens structure with a diameter of 1 µm and a height of 2.5 µm, positioned on a substrate with a thickness of 0.5 µm. The simulation domain of the unit cell measures 2 x 2 x 5 µm.

While I am able to run the simulation successfully using PML boundary conditions, I encounter difficulties when switching the boundary conditions in the x and y directions to periodic settings with k_point = mp.Vector3(). Specifically, the issue arises because the field is only absorbed at the upper and lower boundaries along the z-axis, leaving the x and y directions unaccounted for.

I would like to ask for guidance on how to correctly implement a far-field transformation box that converts the near-field results into a periodic far-field pattern, considering a certain periodicity factor, n.

Is it correct that nperiods must be greater than 1? Would nperiods = 20 be sufficient, or should it be much larger?

I’ve attached a figure to further illustrate the simulation setup. Here I illustrated the possible far field detection possibilities. But I am not sure which one is corrct.

Thank you in advance for your help!

![image](https://github.com/user-attachments/assets/02677fc4-99b4-4e39-a1db-a870ce6f8e0c)

**Comment by oskooi** [comment, marked as answer]
Using the near-to-far field transformation feature for a unit cell with 2D periodic boundaries is discussed in the [manual](https://meep.readthedocs.io/en/latest/Python_Tutorials/Near_to_Far_Field_Spectra/#focusing-properties-of-a-metasurface-lens):

<img width="726" alt="Screenshot 2025-01-09 at 12 25 51" src="https://github.com/user-attachments/assets/798c434f-aa80-445b-a4a7-390d22923b7e" />
